### PR TITLE
4 virt launcher domain spec

### PR DESCRIPTION
**Pull request by cijohnson**
**What this PR does / why we need it**:
1. Creates virsh xml elements required for vhostuser interface by readiing the vhostuser socket file name from the pod annotations file, which is populated via DownwardAPI/userspace CNI and use it in domain spec.
    
2. Exported lookupInterfaceByNetwork and removed findInterfaceByNetworkName.
    Moved/exported composePodInterfaceName to network/vmispec pkg.
    These functions will be used by both virtwrap/converter and network/setup pkgs.
    
3. Vendored following deps,
    github.com/openshift/app-netutil/lib and
    github.com/k8snetworkplumbingwg/network-attachment-definition-client

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Related #3405

**Special notes for your reviewer**:
This PR is  the continuation of https://github.com/kubevirt/kubevirt/pull/3888
This PR addresses the 4th bullet point  in comment https://github.com/kubevirt/kubevirt/issues/3405#issuecomment-659373177 

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by rmohr** [comment]
@cijohnson would you mind giving a high level overview on the PR description? Motivation, general flow, implications, vmi example, ...

**Comment by cijohnson** [comment]
> @cijohnson would you mind giving a high level overview on the PR description? Motivation, general flow, implications, vmi example, ...

@rmohr  updated the description with required information.

**Comment by rmohr** [comment]
@stu-gott @krsacme I have right now no capacity to dive into this topic do you think you can help out here a little bit?

**Comment by rmohr** [comment]
@AlonaKaplan, @phoracek  or since it is related to DPTK, maybe something for networking?

File: pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter_test.go [hunk 1]
```diff
@@ -3479,6 +3479,7 @@ var _ = Describe("Validating VMICreate Admitter", func() {
 			causes := webhooks.ValidateVirtualMachineInstanceArm64Setting(k8sfield.NewPath("spec"), &vmi.Spec)
 			Expect(len(causes)).To(Equal(1))
 		})
+	})
```
Comment (maiqueb) [inline pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter_test.go:3482]: Please move this to the previous commit (which is broken).

**Comment by maiqueb** [comment]
/sig network

First of, thanks for this contribution. I find it very valuable. I'll try to review it asap. 

I have some overall general comments about this feature:
  - I think we should focus first on the dependent PR - #3888 , and only afterwards focus on this one (which changes kubevirt's API).
  - I think this feature should be feature gated, as macvtap is. It has potential to graduate a lot sooner than macvtap since all the code resides in kubevirt.
  - I notice that you're missing e2e tests. AFAIU, we need those to be able to safely accept this feature. This could be tricky, but, at least, we do not require special hardware.

**Comment by poojaghumre** [comment]
Is there any plan to support OVS+DPDK vhostuser interface type in Kubevirt?

**Comment by phoracek** [comment]
@poojaghumre see https://groups.google.com/g/kubevirt-dev/c/opZPTUNsoMQ, that is I believe the latest conversation about the topic

**Comment by fgschwan** [comment]
@cijohnson Hi, i'm working on very similar use case to use case described in the oldest predecessor PR https://github.com/kubevirt/kubevirt/pull/3208 . I want to connect Kubevirt with VPP instead of OVS. This use case is basically the same from Kubevirt perspective. It differs in userspace-cni as it doesn't support for vhost for VPP in the same way as for OVS (it is only done as example with hardcoded stuff). I did describe my use case more in discussion in mentioned PR (https://github.com/kubevirt/kubevirt/pull/3208#issuecomment-970348986) and in Kubevirt mailing thread (https://groups.google.com/g/kubevirt-dev/c/opZPTUNsoMQ/m/nHgRMrsKAgAJ?pli=1).

I based the setup on your [vhost-support Kubevirt version](https://github.com/cijohnson/kubevirt/releases/tag/v0.48.0-jnpr) tha contains this PR and uses more recent version of Kubevirt that the old PRs. I must thank you for that.

Also i'm thinking about releasing some setup for my use case when it starts to work (currently still work in progress). It can help and speed up work of others because currently i'm not aware there are some public examples of this working. However my setup will be probably nothing that can be merged somewhere as nice example because i'm just modifying the Kubevirt development environment setup (Kind setup of k8s+Kubevirt) with a lot of hacky/hardcoded stuff.

**Comment by cijohnson** [comment]
@fgschwan thanks for offering to help. We need help with automating the tests and make it part of kubevirt CI. only then community will approve this change.

@krsacme  was working on the CI e2e tests. if you can help with this, I will setup a meeting with @krsacme  and yourself to get knowledge transfer from @krsacme . Please let me know. Thanks

**Comment by fgschwan** [comment]
@cijohnson Hi, i'm still in the early setup stage of the use case mentioned. First i have to successfully finish the full use case (VPP<-vhost->VM is only part of it) and only then i can help with merging of the vhost support. I'm aware that the CI test is the main roadblock for this to get merged. However i put some effort to setup vhost because there is no example setup available. To prevent future waste of effort for new people wanting to use vhost, it is probably wise to make my effort public when it is finished. I understand that this doesn't help with CI right now however it can still help others.

**Pull request closed without merging**